### PR TITLE
docs(cli): added new documentation for Ignite CLI

### LINKED ISSUE #2291: Documentation: Document the Ignite CLI in a separate doc
We really need a doc (maybe `./docs/Ignite-CLI.md`) where we document all commands and link to specific other pages as needed (like the page for Generators).

### PR DESCRIPTION
**Pull request by frankcalise**
## Describe your PR
- Closes #2291
- Initial go at documenting the power behind the CLI!

image

### CODE REVIEW

File: docs/Ignite-CLI.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+- [remove-demo](#remove-demo)
+- [rename](#rename)
+- [snackify](#snackify)
+- [update](#update)
+
+### Cache
+
+- `npx ignite-cli cache`
+
+Generally speaking, you shouldn't have to use this command. However, it is helpful for managing the dependency cache Ignite uses when spinning up new projects. You can opt out of using the cache, see more information under the [new command](#new).
```
Comment (joshuayoes) [inline docs/Ignite-CLI.md:23]: This to me suggests that the cache is used by default, which it is not. I have some suggestions on a replacement here

```suggestion
The [`new` command](#new) has a `useCache` flag that allows you to cache your dependencies to speed up future uses of `new`. By default, this flag is `false`. This command is provided to allow for interacting with the dependency cache file folder.

#### Subcommands
- `npx ignite-cli cache help` outputs help command to describe cache subcommands
- `npx ignite-cli cache path` outputs the path of the cache folder on your system 
- `npx ignite-cli cache clear` deletes the cache folder on your system
```

File: docs/Ignite-CLI.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+- [issue](#issue)
+- [remove-demo-markup](#remove-demo-markup)
+- [remove-demo](#remove-demo)
+- [rename](#rename)
+- [snackify](#snackify)
+- [update](#update)
+
+### Cache
+
+- `npx ignite-cli cache`
```
Comment (joshuayoes) [inline docs/Ignite-CLI.md:21]:
```suggestion
- `npx ignite-cli cache`
- alias `npx ignite-cli c`
```

File: docs/Ignite-CLI.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+Provides generators to keep your code consistent while saving you time to scaffold new models, components and screens in an automated fashion.
+
+For full documentation on this, head on over to the [Generators documentation](./Generators.md).
+
+### Help
+
+- `npx ignite-cli help`
+- Alias: `npx ignite-cli h`
+
+Provides this list of commands and a small description of each in your terminal.
```
Comment (joshuayoes) [inline docs/Ignite-CLI.md:89]:
```suggestion
Provides this list of commands and a small description of each command in your terminal.
```

File: docs/Ignite-CLI.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+- `--removeDemo` will remove the boilerplate demo code after project creation
+- `--useCache` flag specifying to use dependency cache for quicker installs
+- `--yes` accept all prompt defaults
+
+### Issue
+
+- `npx ignite-cli issue "Mac M1 install trouble"`
+- Alias: `npx ignite-cli i`
+
+Fires up a new issue on GitHub.com prefilled with collected [doctor](#doctor) information. Simply describe your steps to help reproduce the issue (and provide any relevant code snippets or repository) and press submit!
```
Comment (joshuayoes) [inline docs/Ignite-CLI.md:114]:
```suggestion
Fires up a [new issue for Ignite on GitHub](https://github.com/infinitered/ignite/issues/new/) prefilled with collected [doctor](#doctor) information. Simply describe your steps to help reproduce the issue (and provide any relevant code snippets or repository) and press submit!
```

**Review by joshuayoes — CHANGES_REQUESTED** [review]
I've got some nitpicky grammar suggestions and an edit of the cache command section. But otherwise, this looks great! This will be really useful for Ignite users.

**Comment by frankcalise** [comment]
> I've got some nitpicky grammar suggestions and an edit of the cache command section. But otherwise, this looks great! This will be really useful for Ignite users.

Thanks appreciate it!

**Review by jamonholmgren — APPROVED** [review]
This is :100: excellent work

**Pull request merged**